pkg/apis/policy/v1alpha1/propagation_types.go

```diff
@@ -70,6 +70,16 @@ type PropagationSpec struct {
 	// +optional
 	Placement Placement `json:"placement,omitempty"`
 
+	// Priority indicates the importance of a policy(PropagationPolicy or ClusterPropagationPolicy).
+	// A policy will be applied for the matched resource templates if there is
+	// no other policies with higher priority at the point of the resource
+	// template be observed.
+	// Once a resource template has been claimed by a policy, it can not be
+	// preempted by following polices even with a higher priority.
+	// The higher the value, the higher the priority. Defaults to zero.
+	// +optional
+	Priority *int32 `json:"priority,omitempty"`
```

Comment:
Why is `*int32` type used? Are there any restrictions on using `int32`?

Reply:
Echo from [API-conversion]():
> They(optinal fileds) are a pointer type in the Go definition (e.g. AwesomeFlag *SomeFlag) or have a built-in nil value (e.g. maps and slices).

But I also realized each time when access this field, we should check if it is nil, that's pretty annoying, especially when we can definitely be sure it can't be nil(since it has a default value).

So, I think we can use `int32` for this case.

Reply:
@jwcesign Please start to implement it based on this version. Let's see the type is an annoying thing.

Reply:
So what about `Priority int32 `json:"priority"`?

Reply:
After going through the implementation(#2767), looks not so bad. 

In addition, we can add some helpers function for the API, like:
```go
func (p *PropagationSpec) ExplicitPriority() int32 {
	if p.Priority == nil {
		return 0
	}
	return *p.Priority
}
```
So, @Garrybest do you have other comments?

Reply:
I'm ok with this.